app/controllers/results_submission_controller.rb

```diff
@@ -127,15 +133,9 @@ def import_from_live
                 end
     end
 
-    if errors.any?
-      flash[:danger] = errors.join("<br/>")
-      @results_validator = ResultsValidators::CompetitionsResultsValidator.create_full_validation
-      @results_validator.validate(@competition.id)
-      return render :new
-    end
+    return render status: :unprocessable_entity, json: { error: errors.join("<br/>") } if errors.any?
```

Comment:
Why do you need to join the errors with `<br>` tags here already? This was necessary in the old world because it was a `flash` message (which cannot be dynamic).

Now you're moving this to the new React world, where you can just return the individual error messages in a JSON array, and then have React do whatever it wants with this list (for example, loop over them in a `<List>`, or whatver...)

Reply:
Will attend this soon.

Reply:
Cool!